tests/test_web_urldispatcher.py

```diff
@@ -813,19 +830,21 @@ async def test_static_absolute_url(
     # where the static dir is totally different
     app = web.Application()
     file_path = tmp_path / "file.txt"
-    file_path.write_text("sample text", "ascii")
+    await asyncio.to_thread(file_path.write_text, "sample text", "ascii")
     here = pathlib.Path(__file__).parent
-    app.router.add_static("/static", here)
+    await asyncio.to_thread(app.router.add_static, "/static", here)
     client = await aiohttp_client(app)
-    resp = await client.get("/static/" + str(file_path.resolve()))
+    resp = await client.get(
+        "/static/" + str(await asyncio.to_thread(file_path.resolve))
+    )
     assert resp.status == 403
 
 
 async def test_for_issue_5250(
     aiohttp_client: AiohttpClient, tmp_path: pathlib.Path
 ) -> None:
     app = web.Application()
-    app.router.add_static("/foo", tmp_path)
+    await asyncio.to_thread(app.router.add_static, "/foo", tmp_path)
```

Comment:
This seems like a bug in protection code that add_static does blocking IO.  I'm pretty sure Home Assistant has a workaround for this so it would be good to fix in the production code

Reply:
https://github.com/home-assistant/core/blob/634b754168c19a51b72079749cccdceec5551aea/homeassistant/components/http/__init__.py#L509

Reply:
https://github.com/home-assistant/core/blob/634b754168c19a51b72079749cccdceec5551aea/homeassistant/components/http/__init__.py#L477

Reply:
Given that this can't be done once the app is frozen (other than homeassistant hacking it), it's probably not been an issue so far.

Reply:
The blocking detected is through the path `add_static` -> `StaticResource.__init__` which does OS  calls to resolve the directory:
```python
        try:
            directory = Path(directory).expanduser().resolve(strict=True)
        except FileNotFoundError as error:
            raise ValueError(f"'{directory}' does not exist") from error
        if not directory.is_dir():
            raise ValueError(f"'{directory}' is not a directory")
```
I think in general `add_static` is called outside of an event loop. Also the `add_static` doc says it is not recommended for production code. So probably not a big issue if it blocks ?

Reply:
If this is something we should fix (even if homeassistant were the only use case), then should we revert the to_thread() changes and whitelist the issue until fixed?

Reply:
> If this is something we should fix (even if homeassistant were the only use case), then should we revert the to_thread() changes and whitelist the issue until fixed?

done in https://github.com/aio-libs/aiohttp/pull/10433/commits/b7cec65d06a678f88ee627e8b5dd07fe64f41fe7